cannon/mipsevm/witness.go

```diff
@@ -63,6 +66,7 @@ func (wit *StepWitness) EncodePreimageOracleInput() ([]byte, error) {
 		var input []byte
 		input = append(input, LoadLocalDataBytes4...)
 		input = append(input, wit.PreimageKey[:]...)
+		input = append(input, uint32ToBytes32(uint32(localContext))...) // local context in bytes
```

Comment:
Why are we using a `uint64` that we then case to `uint32`? Seems like we should declare it as `uint32` for accuracy.  Converting from `big.Int` will be slightly more difficult but that's where we should be bounds checking the conversion anyway.